src/modules/m_sqloper.cpp

```diff
@@ -121,40 +144,47 @@ class ModuleSQLOper : public Module
 		else
 			SQL.SetProvider("SQL/" + dbid);
 
-		hashtype = tag->getString("hash");
-		query = tag->getString("query", "SELECT hostname as host, type FROM ircd_opers WHERE username='$username' AND password='$password' AND active=1;");
+		query = tag->getString("query", "SELECT * FROM ircd_opers WHERE active=1;");
+		const std::string noexist = "null"; //Because OpGetQuery never returns something to ModuleSQLOper, it is resposible for executing /OPER if necessary. Therefore, it requires params such as username and pwd that aren't necessary if it's updating for /REHASH.
+
+		GetOLines(false, noexist, noexist, noexist); //Update sqloper list from the database.
+		return;
+	}
+
+	CullResult cull()
+	{
+		ServerInstance->Config->oper_blocks = conf_blocks; //Reset oper blocks to those from the config
+		return Module::cull();
 	}
 
 	ModResult OnPreCommand(std::string &command, std::vector<std::string> &parameters, LocalUser *user, bool validated, const std::string &original_line) CXX11_OVERRIDE
 	{
-		if (validated && command == "OPER" && parameters.size() >= 2)
+		if (validated && command == "OPER" && parameters.size() >= 2 && original_line != "SQLOPER") //The line will be SQLOPER only if this module already reloaded the DB and is recalling /oper for other hooks
 		{
 			if (SQL)
 			{
-				LookupOper(user, parameters[0], parameters[1]);
-				/* Query is in progress, it will re-invoke OPER if needed */
-				return MOD_RES_DENY;
+				GetOLines(true, user->uuid, parameters[0], parameters[1]); //Update opers from DB
+				return MOD_RES_DENY; //We need to reload oper blocks from the DB before other hooks can run (i.e. sslinfo). We will re-call /OPER later.
 			}
 			ServerInstance->Logs->Log(MODNAME, LOG_DEFAULT, "database not present");
 		}
-		return MOD_RES_PASSTHRU;
+		return MOD_RES_PASSTHRU; //There is either no DB or we successfully reloaded oper blocks
 	}
 
-	void LookupOper(User* user, const std::string &username, const std::string &password)
+	void GetOLines(bool OnUserCommand, const std::string u, const std::string &un, const std::string &pw)
```

Comment:
OLines is a really archaic term. Can you use something more appropriate like OperBlocks?

Also, when modifying lines where the &* is against the variable name you can move them to be against the type name.